### PR TITLE
Add webui2 version of kiwi images

### PR DESCRIPTION
**Pull request by Ana06**
image

image

image

image

Supersedes https://github.com/openSUSE/open-build-service/pull/6170

Remember that by now we don't want to refactor/get ride of the horrible JavaScript. It was like that in the old UI and by now we just migrate it and we will refactor if afterwards.

### CODE REVIEW

**Comment by Ana06** [comment]
[Kiwi Editor: Details & Software tab](https://trello.com/c/ar1u8PgJ/1104-kiwi-editor-details-software-tab)

File: src/api/app/views/webui2/webui/kiwi/images/_package_fields.html.haml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      #flash-messages
+        %p.ui-state-error.alert-danger.p-3.mb-0.d-none
+          The name can not be empty!
+
+      .modal-body
+        %p
+          = render partial: 'webui/autocomplete', locals: { html_id: :name, label: 'Name:'.html_safe,
+                                                              source: url_for(controller: '/webui/kiwi/images', action: 'autocomplete_binaries'),
+                                                              default: f.object.name }
```
Comment (coolo) [inline src/api/app/views/webui2/webui/kiwi/images/_package_fields.html.haml]: the variable is called `value` not `default` - and as such the dialog misses a required value and as such rails abort the form submission

Comment (Ana06) [reply src/api/app/views/webui2/webui/kiwi/images/_package_fields.html.haml]: Thanks for the help @coolo :cupid:

**Comment by Ana06** [comment]
[Kiwi Editor: Overview tab](https://trello.com/c/OAQnoFUL/1105-kiwi-editor-overview-tab)

**Comment by Ana06** [comment]
@hellcp JYFI I have squashed all your commits from https://github.com/openSUSE/open-build-service/pull/6170 into b717497  and made several changes:

- Fix saving (as @coolo pointed out in [his comment](https://github.com/openSUSE/open-build-service/pull/7565#discussion_r284390402) the partial variable name was wrong which blocked the sending of the form)
- Fix cancel button in modals which was not working
- Make buttons consistent with other views within the new UI (I know that there are already some inconsistency, but I took what we have in the share partials as the default)
- Remove intrusive Javascript
- move Javascript to assets (so that it is precompile)
- Fix Jshint offenses
- Rebase and stop using `User.current` which doesn't exist anymore
- Move `switch_to_webui2` to the html reponse
- Correct use of `fields_for`

I also add two extra commits with other independent changes. In case you want more details or something is not clear, just let me know. :wink: Thanks for the help with the bootstrap migration @hellcp! :cupid:

**Comment by DavidKang** [comment]
@Ana06, I rebased with master and fixed the conflicts

**Comment by hennevogel** [comment]
:clap: :clap: :clap:

**Pull request merged**